policyengine_us/variables/gov/states/hi/tax/income/deductions/itemized/hi_interest_deduction.py

```diff
@@ -0,0 +1,32 @@
+from policyengine_us.model_api import *
+
+
+class hi_interest_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii interest deduction"
+    unit = USD
+    documentation = (
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=17"
+        "https://files.hawaii.gov/tax/forms/2022/n11ins.pdf#page=32"  # total itemized deduction worksheet
+    )
+    definition_period = YEAR
+    defined_for = StateCode.HI
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.irs.deductions.itemized.interest.mortgage
+
+        # 3. interest_deduction: worksheet A-3
+        # Hawaii did not
+        #     (1) suspend the deduction for interest paid on home equity loans
+        #     (2) lower the dollar limit on mortgages qualifying for the home mortgage interest deduction
+        filing_status = tax_unit("filing_status", period)
+        home_mortgage_interest = min_(
```

Comment:
The cap applies to the mortgage value, not the interest payments. We can't easily model this, so we assume that mortgage interest applies only to the relevant part of the mortgage.
```suggestion
        home_mortgage_interest = add(tax_unit, period, ["mortgage_interest"])
```
For future reference though, would suggest breaking this kind of logic into three statements, one for the amount, one for the cap, one for the capped amount.

Reply:
then it seems like we can just use `adds = ["investment_interest_expense", "mortgage_interest"]` here as opposed to a formula if there is no cap